CIPs/0008.md

```diff
@@ -64,14 +63,56 @@ I considered doing storing all data on IPFS, however the tooling around IPFS as
 
 More reliance on off-chain data and misunderstanding such reliance could lead to inconsistency or poor usability. Ultimately this is traded off against cost of storing data on-chain or in centralized services.
 
-## Useful Links
+## Specification
+
+### Signatures
+
+All file contents should always be signed by an authorized signer of the account to prevent tampering by the file hoster. Signatures should be stored along side the payload under `${dataPath}.signature` and to construct these signatures the signature specification [EIP712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md) is used. Where it doesn’t make sense to include the payload itself in the EIP712 object we instead include the keccak256 hash hex encoded.
```

Comment:
```suggestion
All file contents should always be signed by an authorized signer of the account to prevent tampering by the file hoster. Client may reject any unsigned data received from the storage root. Signatures should be stored along side the payload under `${dataPath}.signature` and to construct these signatures the signature specification [EIP712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md) is used. Where it doesn’t make sense to include the payload itself in the EIP712 object we instead include the keccak256 hash hex encoded.
```